libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -310,13 +312,14 @@ object MapboxRouteLineUtils {
      */
     fun calculateRouteLineSegments(
         route: DirectionsRoute,
+        congestionRange: CongestionRange,
         trafficBackfillRoadClasses: List<String>,
         isPrimaryRoute: Boolean,
         routeLineColorResources: RouteLineColorResources,
         restrictedRoadSectionScale: Double,
         displayRestrictedRoadSections: Boolean
     ): List<RouteLineExpressionData> {
-        val trafficExpressionData = getRouteLineTrafficExpressionDataFromCache(route)
+        val trafficExpressionData = getRouteLineTrafficExpressionDataFromCache(route, congestionRange)
```

Comment:
This can't change. The cache is dependent on the arguments. If you look at the data structure it's retuning, the `RouteLineTrafficExpressiondata` it's really just extracting needed data from the route. You can replace the current 'trafficCongestionIdentifier: String` with the congestion range.  That data is then passed to `getRouteLineExpressionDataWithStreetClassOverride`. Inside that function there is some logic to get the traffic identifier, line 542. Make the decision there based on the range rather than where you're currently making that decision.

Reply:
> You can replace the current trafficCongestionIdentifier: String with the congestion range

I don't think I can do that since we need to make it backwards compatible with `congestion`. If while making the route request the user passes `DirectionsCriteria.ANNOTATION_CONGESTION`, then there will be no `CongestionRange` and just `String`. This data structure would then fail.